server/proxy/grpcproxy/cluster.go

```diff
@@ -79,38 +74,26 @@ func NewClusterProxy(lg *zap.Logger, c *clientv3.Client, advaddr string, prefix
 	return cp, donec
 }
 
-func (cp *clusterProxy) resolve(prefix string) {
-	rm := rate.NewLimiter(rate.Limit(resolveRetryRate), resolveRetryRate)
-	for rm.Wait(cp.ctx) == nil {
```

Comment:
So, this deleted part is being moved to `client/v3/naming/endpoints`?

Reply:
I don't think there is anything worth to move in the resolve method.

Please notice that cp.gr.Resolve(prefix) was always succeeding: 

```
func (gr *GRPCResolver) Resolve(target string) (naming.Watcher, error) {	
	ctx, cancel := context.WithCancel(context.Background())	
	w := &gRPCWatcher{c: gr.Client, target: target + "/", ctx: ctx, cancel: cancel}	
	return w, nil	
}
```

So there is no value in the throttling / retry loop. 
That would be needed if we wanted to wait for not empty set of servers..., but it's not the current semantic.

Reply:
Added more explicit error handling.